### ISSUE TITLE
Support reading ETC2 (RGB)

### ISSUE THREAD

**Issue by BaalNetbek** [Trimatix/AEPi#40]
Trying to open textures from GoF3D for Symbian I get an error of unknown compression format. The AEIs can't be opened also with Catlabs AEI Editor.

[textures.zip](https://github.com/Trimatix/AEPi/files/15225260/textures.zip)

```
Exception in Tkinter callback
Traceback (most recent call last):
  File "C:\Users\user\AppData\Roaming\Python\Python311\site-packages\AEPi\image\AEI.py", line 305, in read
    format, mipmapped = CompressionFormat.fromBinary(formatId)
                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Roaming\Python\Python311\site-packages\AEPi\constants.py", line 40, in fromBinary
    raise exceptions.InvalidCompressionFormatException(rawId)
AEPi.exceptions.InvalidCompressionFormatException: The AEI could not be read. The AEI specified an unknown compression format: 23.

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "X:\Program Files\Python311\Lib\tkinter\__init__.py", line 1948, in __call__
    return self.func(*args)
           ^^^^^^^^^^^^^^^^
  File "R:\image 5g.py", line 222, in preview_image
    with AEI.read(image_path) as aei:
         ^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Roaming\Python\Python311\site-packages\AEPi\image\AEI.py", line 351, in read
    raise AeiReadException(None, ex) from ex
AEPi.exceptions.AeiReadException: The AEI could not be read.
The AEI could not be read. The AEI specified an unknown compression format: 23.
```

**Comment by Trimatix** [comment]
Many thanks for the report. Format 23 doesn't fit with the usual format IDs of GOF2 AEIs for a couple of reasons, were these texture files taken directly from a working, original installation of GOF3D?
- Format 23 seems to represent ETC2 + alpha (see #41), but the AEI format ID for ETC1 is 64 - a big jump! Different versions of the same format are usually close-ish together: DXT1 = 32, DXT3 = 33, DXT5 = 36
- ID 23 requires the mipmapping flag (2) to be high, so we don't know whether or not images in this format are stored with mipmapping.

Regardless, AEPi is able to successfully decompress your provided textures using the existing Tex2Img codec adaptor. This isn't a requirement for V1, but I'll publish a new version that adds support anyway!

**Comment by Trimatix** [comment]
This feature has been released in AEPi 0.8.3 and is available on pypi. Since it's not part of the ""official"" pool of formats (i.e supported by AEIEditor), I'm leaving it untested and undocumented for now. I've also only implemented decompression and not compression; compression will take a bit more work, because we'll need to find a codec for it. I've scheduled that work, and the same for the other formats I've discovered (#41) for "post V1", since it's not required to meet feature parity with AEIEditor.

If anyone else would find more support for this format (or others from #41) useful, forking and PR'ing would be hugely appreciated!

**Comment by BaalNetbek** [comment]
Are you sure it is ETC2 or mabey it's just bugly implemented? It doesn't work well for me. 
Using such simple script many times it never reads the subject .aei. 
```
from AEPi import AEI
    
aei_file = "main_texture.aei"
    
with AEI.read(aei_file) as aei:
    aei.getTexture().save(f"{aei_file}.png")
    for i, tex in enumerate(aei.textures):
        with aei.getTexture(tex) as im:
            im.save(f"{aei_file}_{i}.png")  
```
And using the AEIPorter it sometimes export a PNG sometimes just crushes and these. The exported images are very deformed vary by lower half which is what seems to be random artifacts. I don't think these AEIs of interest do even have alpha channel.

example exported
![obraz](https://github.com/Trimatix/AEPi/assets/157546829/e2ef7555-9334-472a-81b8-a7eee0e13f99)
![main_texture_0](https://github.com/Trimatix/AEPi/assets/157546829/099b8c38-e53a-4ad1-8fe2-e1ddf8fd5701)

desired texture (I ripped it from running game)
![obraz](https://github.com/Trimatix/AEPi/assets/157546829/c436752e-5f08-4ec7-9a8c-f3919c407a54)

**Comment by BaalNetbek** [comment]
So it look like it's not RGBA but BGR.

**Comment by Trimatix** [comment]
Apologies, you're absolutely right! That's what i get for not writing unit tests i suppose 🙃

I've retested using all of your files, and the results are the same now as what you're expecting.
What's really interesting is that, even though they're compressed with ETC, the channels are stored as RGB instead of BGR! I'll make a note of the new info in #41 as well.

I'm publishing this fix now to v0.8.3.1 on pypi :)

**Comment by BaalNetbek** [comment]
Awesome
